multi-output version

rfiorella opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

rfiorella committed:
test multi-output version

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found some lint.

Here's what I've got...


For **recipe/meta.yaml**:

 * ❌ The top level meta keys are in an unexpected order. Expecting ['package', 'source', 'build', 'requirements', 'test', 'outputs', 'about', 'extra'].


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/18077979028](https://github.com/conda-forge/conda-forge-webservices/actions/runs/18077979028). Examine the logs at this URL for more detail.</sub>

rfiorella committed:
add pytrilinos cmake flag

rfiorella committed:
fix linter issue

rfiorella committed:
add scikit-umfpack constraint

rfiorella committed:
remove scikit constraint, add numpy

rfiorella committed:
pin numpy

rfiorella commented:
@conda-forge-admin, please rerender.

Commit:
MNT: Re-rendered with conda-smithy 3.52.2 and conda-forge-pinning 2025.09.27.05.44.0

rfiorella commented:
@conda-forge-admin, please restart ci

rfiorella committed:
Update meta.yaml

Remove numpy constraints

rfiorella committed:
avoid free-threaded issue

rfiorella committed:
fix selector

rfiorella committed:
Update meta.yaml

Add suitesparse for missing libraries

Commit:
MNT: Re-rendered with conda-smithy 3.52.3 and conda-forge-pinning 2025.10.17.17.13.4

rfiorella committed:
replace scikit-umfpack with suitesparse

rfiorella committed:
try explicitly specifying umfpack dirs

rfiorella committed:
add swig

rfiorella committed:
wip - debug pytrilinos build error

h-vetinari commented:
Even though it's a very old PR, you may be able to get some inspiration from https://github.com/conda-forge/pytrilinos-feedstock/pull/12

rfiorella committed:
tweak multi-output after mamba-feedstock

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * ℹ️ The recipe is not parsable by parser `conda-souschef (grayskull)`. This parser is not currently used by conda-forge, but may be in the future. We are collecting information to see which recipes are compatible with grayskull.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/19519106257](https://github.com/conda-forge/conda-forge-webservices/actions/runs/19519106257). Examine the logs at this URL for more detail.</sub>

rfiorella committed:
clarify requirements across outputs

rfiorella committed:
wip - add debug statement. swig can't find teuchos headers

rfiorella committed:
MNT: Re-rendered with conda-smithy 3.53.3 and conda-forge-pinning 2025.12.04.18.32.37

Other tools:
- conda-build 25.11.0
- rattler-build 0.53.0
- rattler-build-conda-compat 1.4.10

rfiorella committed:
refactor multi-output build

rfiorella committed:
try explicit enable of pytrilinos packages

rfiorella committed:
add some missing dependencies

rfiorella committed:
specify paths for swig

rfiorella committed:
tweak recipe for pytrilinos

rfiorella committed:
attempt to patch swig include errors

rfiorella committed:
fix missing trilinos requirements for pytrilinos output

rfiorella committed:
fix BSD sed, pytrilinos import issues

rfiorella committed:
pytrilinos test needs more run exports

rfiorella committed:
remove domi, removed in earlier version of trilinos

rfiorella committed:
remove more deprecated/removed packages

rfiorella commented:
@conda-forge-admin please rerender.

Commit:
MNT: Re-rendered with conda-smithy 3.54.2 and conda-forge-pinning 2026.02.21.19.20.5

rfiorella committed:
add numpy > 2 constraint

rfiorella committed:
MNT: Re-rendered with conda-smithy 3.55.1 and conda-forge-pinning 2026.02.21.19.20.56

Other tools:
- conda-build 26.1.0
- rattler-build 0.57.2
- rattler-build-conda-compat 1.4.11

rfiorella committed:
forgot to remove Komplex from PyTrilinos tests

rfiorella committed:
add cmake patch for python >= 3.10

rfiorella committed:
other places where python version not being correctly extracted

rfiorella committed:
tweak pytrilinos patch

rfiorella committed:
still getting truncated python version

rfiorella committed:
specify more paths

rfiorella committed:
simpler attempt to get pytrilinos paths right

rfiorella committed:
swig include patch: glob all packages/*/src* dirs

rfiorella committed:
fix hdf5 link: pass explicit lib paths to bypass cmake target name

rfiorella committed:
swig: add packages/*/src/* glob for nested header dirs

rfiorella committed:
swig: add packages/*/*/compat glob for generated config headers

rfiorella committed:
swig: replace targeted glob with recursive header dir walk

rfiorella committed:
swig: target UseSWIG.cmake directly, don't pollute compile -I list

rfiorella committed:
pin hdf5 >=2.0 in trilinos host+run to keep test env on built ABI

rfiorella committed:
pin TPL paths for HDF5, NetCDF, Matio, Boost

Trilinos's TriBITS finders auto-detect TPLs via find_library, which on
macOS conda-forge picks up libhdf5.so symlinks instead of .dylib and
bakes the wrong path into the ninja build. Set TPL_*_LIBRARIES and
TPL_*_INCLUDE_DIRS explicitly using ${SHLIB_EXT} so the same flags work
on both Linux and macOS and the finder never runs.

rfiorella committed:
remove duplicate TPL_HDF5_LIBRARIES override

The hardcoded .so paths added during macOS debugging shadowed the
${SHLIB_EXT} flag earlier on the cmake line (last -D wins for cache
entries), baking libhdf5.so / libhdf5_hl.so into the ninja rules
on macOS where the conda-forge hdf5 package only ships .dylib.

Commit:
MNT: Re-rendered with conda-smithy 2026.5.29 and conda-forge-pinning 2026.06.08.09.22.5

rfiorella committed:
Merge branch 'main' into multi-output-recipe

rfiorella committed:
fix linter suggestions

rfiorella committed:
fix linter numpy issue

rfiorella commented:
thanks, I'll submit the request as soon as I can

h-vetinari commented on recipe/conda_build_config.yaml:
```diff
@@ -1,3 +1,5 @@
 mpi:
   - openmpi
   - mpich
+numpy:
+  - 2 
```
this is redundant with what the pinning already sets

h-vetinari commented on recipe/meta.yaml:
```diff
@@ -21,36 +27,106 @@ requirements:
-    - pytrilinos ==999999999
+    - libnetcdf
+    - netcdf-cxx4
+    - python
+    - python_abi * *_cp313  # [py==313]
+    - numpy
```
this is all covered by run-exports; more to the point, this `run:` section isn't actually attached to any output.

h-vetinari commented on recipe/meta.yaml:
```diff
@@ -21,36 +27,106 @@ requirements:
     - libmatio
     - xorg-libx11
-    - hdf5 * mpi_{{ mpi }}_*
+    - hdf5 >=2.0 mpi_{{ mpi }}_*
```
This should be
```suggestion
    - hdf5
    - hdf5 * mpi_{{ mpi }}_*
```
and then use the migration infrastructure (e.g. copying [this](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/main/recipe/migrations/hdf52.yaml) under `.ci_support/migrations`).

h-vetinari reviewed (commented):
I would have expected another round of review for such a big change (I wanted to see green CI before reviewing)...

Overall this looks pretty OK though.

h-vetinari commented:
It's no a big deal, these are comparatively minor things that can be fixed in follow-ups; I just would have expected you to ask for review before merging, given the whole history of this trilinos saga. In any case, congrats for sticking to it and getting it unstuck. :)
